addons/test_mail_full/static/tests/tours/message_actions_tour.js

```diff
@@ -12,3 +13,75 @@ registry.category("web_tour.tours").add("star_message_tour", {
         },
     ],
 });
+
+registry.category("web_tour.tours").add("message_actions_tour", {
+    steps: () => [
+        {
+            trigger: "#chatterRoot:shadow .o-mail-Thread .o-mail-Message",
+            run: async function () {
+                await contains(".o-mail-Thread .o-mail-Message", {
+                    count: 1,
+                    target: document.querySelector("#chatterRoot").shadowRoot,
+                });
+            },
```

Comment:
Why do we have to repeat the selector inside the run?
We try to avoid custom run and rely on tour steps whenever possible.

Reply:
I need to count the number of messages in different steps. This isn't possible with a simple contains inside the trigger.

Reply:
Ok for now, it would be best to introduce a pseudo selector for counting eventually if we don't have it yet.